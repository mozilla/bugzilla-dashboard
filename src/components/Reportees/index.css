.highlight {
    color: #e53935;
    font-weight: bold;
}
.reportees-table a {
    text-decoration: none;
    /* color: #000000de; */
}
.reportees-table th {
    text-align: center;
}