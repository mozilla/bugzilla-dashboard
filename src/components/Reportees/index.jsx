import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import ExpandMore from '@material-ui/icons/ExpandMore';

const styles = ({
  root: {
    margin: '0 0.5rem 0 0',
  },
  header: {
    margin: '0.5rem 0 0.5rem 0',
  },
  icon: {
    fontSize: '1rem',
    verticalAlign: 'bottom',
  },
  person: {
    // This makes it line up better with the table of components
    padding: '1px',
  },
});

const sortByPersonName = (a, b) => (a.cn <= b.cn ? -1 : 1);

const Reportees = ({
  classes, ldapEmail, partialOrg, onPersonDetails,
}) => (
  <div className={classes.root}>
    <h3 className={classes.header}>Reportees</h3>
    <div height="1rem">&nbsp;</div>
    {Object.values(partialOrg)
      .filter(({ cn }) => cn !== ldapEmail)
      .sort(sortByPersonName)
      .map(({ cn, mail }) => (
        <div key={mail} className={classes.person}>
          <span
            name={mail}
            onKeyPress={e => onPersonDetails(e, mail)}
            onClick={e => onPersonDetails(e, mail)}
            role="button"
            tabIndex="0"
          >
            <ExpandMore classes={{ root: classes.icon }} />
          </span>
          <span>{`${cn} `}</span>
        </div>
      ))}
  </div>
);

Reportees.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  ldapEmail: PropTypes.string.isRequired,
  partialOrg: PropTypes.shape({}).isRequired,
  onPersonDetails: PropTypes.func.isRequired,
};

export default withStyles(styles)(Reportees);
