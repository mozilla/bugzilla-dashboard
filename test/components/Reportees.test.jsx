import React from 'react';
import renderer from 'react-test-renderer';
import Reportees from '../../src/components/Reportees';
import partialOrg from '../mocks/partialOrg';

it('renders Someone with no reportees', () => {
  const tree = renderer
    .create((
      <Reportees
        ldapEmail="someone@mozilla.com"
        partialOrg={partialOrg}
        onPersonDetails={() => null}
      />
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Manager who has reportees', () => {
  const tree = renderer
    .create((
      <Reportees
        ldapEmail="manager@mozilla.com"
        partialOrg={partialOrg}
        onPersonDetails={() => null}
      />
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
