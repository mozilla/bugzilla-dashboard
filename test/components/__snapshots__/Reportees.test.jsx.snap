// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Manager who has reportees 1`] = `
<div
  className="Reportees-root-1"
>
  <div
    height="1rem"
  >
     
  </div>
  <div
    className="Reportees-person-4"
  >
    <div
      className="DrilldownIcon-svgWrapper-5"
      name="manager@mozilla.com"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-7 DrilldownIcon-icon-6"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
    <span>
      Manager 
    </span>
  </div>
  <div
    className="Reportees-person-4"
  >
    <div
      className="DrilldownIcon-svgWrapper-5"
      name="someone@mozilla.com"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-7 DrilldownIcon-icon-6"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
    <span>
      Someone 
    </span>
  </div>
</div>
`;

exports[`renders Someone with no reportees 1`] = `
<div
  className="Reportees-root-1"
>
  <div
    height="1rem"
  >
     
  </div>
  <div
    className="Reportees-person-4"
  >
    <div
      className="DrilldownIcon-svgWrapper-5"
      name="manager@mozilla.com"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-7 DrilldownIcon-icon-6"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
    <span>
      Manager 
    </span>
  </div>
  <div
    className="Reportees-person-4"
  >
    <div
      className="DrilldownIcon-svgWrapper-5"
      name="someone@mozilla.com"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-7 DrilldownIcon-icon-6"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
    <span>
      Someone 
    </span>
  </div>
</div>
`;
